src/io/flutter/logging/tree/DataPanel.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2019 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.logging.tree;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.intellij.ui.ColoredTreeCellRenderer;
+import com.intellij.ui.components.panels.NonOpaquePanel;
+import com.intellij.ui.tree.BaseTreeModel;
+import com.intellij.util.EventDispatcher;
+import com.intellij.util.ui.JBUI;
+import io.flutter.inspector.DiagnosticLevel;
+import io.flutter.inspector.DiagnosticsNode;
+import io.flutter.inspector.DiagnosticsTreeStyle;
+import io.flutter.utils.JsonUtils;
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.*;
+import javax.swing.event.HyperlinkEvent;
+import javax.swing.text.html.HTMLEditorKit;
+import java.awt.*;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.EventListener;
+import java.util.List;
+
+import static io.flutter.utils.HtmlBuilder.*;
+
+@SuppressWarnings("ALL")
+public class DataPanel extends JPanel {
+
+  public static interface ContentListener extends EventListener {
+    void hasContent(boolean hasContent);
+  }
+
+  /*
+   * Swing's support for CSS is limited and doesn't allow for multiple classes so we cons
+   * together css styling as a single style attribute.
+   *
+   * These values are loosely derived from devtools/inspector.css.
+   *
+   * TODO(pq): add theme-awareness.
+   */
+  private static class CssStyle {
+    static String forNode(DiagnosticsNode node) {
+      return attrs("style", forLevel(node.getLevel()) + forStyle(node.getStyle()));
+    }
+
+    static String forLevel(DiagnosticLevel level) {
+      switch (level) {
+        case info:
+          return "padding-left: 16px; ";
+        case error:
+          return "color: rgb(244, 67, 54); padding-left: 16px; ";
+        case hint:
+          return "background-color: #fafad2; padding: 8px 8px 8px 24px; margin: 5px 0; ";
+        default:
+          return "";
+      }
+    }
+
+    static String forStyle(DiagnosticsTreeStyle style) {
+      switch (style) {
+        case error:
+          return "background-color: #f97c7c; padding: 12px; margin:5px 0; ";
+        default:
+          return "";
+      }
+    }
+  }
+
+  class NodeRenderer extends ColoredTreeCellRenderer {
```

Comment:
you should use the multi icon tree cell renderer from the inspector instead. It allows multiple icons per row which is important to accurately display inspector diagnostics.

Reply:
Good call.  Done.